rotkehlchen/inquirer.py

```diff
@@ -558,18 +559,31 @@ def find_curve_pool_price(
         """
         assert self._ethereum is not None, 'Inquirer ethereum manager should have been initialized'  # noqa: E501
 
-        pools = get_curve_pools()
-        if lp_token.evm_address not in pools:
+        pool_addr_in_cache = GlobalDBHandler().get_general_cache_value(
+            cache_type=GeneralCacheType.CURVE_POOL_ADDR,
+            name=lp_token.evm_address,
+        )
+        if pool_addr_in_cache is None:
+            return None
+        pool_addr = string_to_evm_address(pool_addr_in_cache)
+        pool_tokens_addrs_in_cache = GlobalDBHandler().get_general_cache_value(
+            cache_type=GeneralCacheType.CURVE_POOL_TOKENS,
+            name=pool_addr,
+        )
+        if pool_tokens_addrs_in_cache is None:
+            # should never happen
+            log.error(f'Found no tokens for curve pool {pool_addr} which is in cache.')
             return None
-        pool = pools[lp_token.evm_address]
+        pool_tokens_addrs = pool_tokens_addrs_in_cache.split(',')
```

Comment:
Ah I just now realized how you are saving the pool tokens in the DB. Wouldn't it be better to change this and use the DB, instead of saving a string of comma separated values? This is not how a database should be used.

Instead use the `cache_type` and `name` column and save multiple values for it!!!! So the primary key becomes, type, name, value